components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/WorkflowConfigHolder.java

```diff
@@ -0,0 +1,264 @@
+            properties = workflowConfig.getProperty();
+            if (properties != null) {
+                loadProperties(properties, workFlowExecutor);
+            }            
+            workflowExecutorMap.put(WorkflowConstants.WF_TYPE_AM_SUBSCRIPTION_DELETION, workFlowExecutor);
+            
+            // Load production token generation workflow configurations
+            workflowConfig = config.getProductionApplicationRegistration();
+            executorClass = workflowConfig.getExecutor();
+            clazz = WorkflowConfigHolder.class.getClassLoader().loadClass(executorClass);
+            workFlowExecutor = (WorkflowExecutor) clazz.newInstance();
+            properties = workflowConfig.getProperty();
+            if (properties != null) {
+                loadProperties(properties, workFlowExecutor);
+            }
+            workflowExecutorMap.put(WorkflowConstants.WF_TYPE_AM_APPLICATION_REGISTRATION_PRODUCTION, workFlowExecutor);
+            
+            // Load sandbox token generation workflow configurations
+            workflowConfig = config.getProductionApplicationRegistration();
+            executorClass = workflowConfig.getExecutor();
+            clazz = WorkflowConfigHolder.class.getClassLoader().loadClass(executorClass);
+            workFlowExecutor = (WorkflowExecutor) clazz.newInstance();
+            properties = workflowConfig.getProperty();
+            if (properties != null) {
+                loadProperties(properties, workFlowExecutor);
+            }
+            workflowExecutorMap.put(WorkflowConstants.WF_TYPE_AM_APPLICATION_REGISTRATION_SANDBOX, workFlowExecutor);
+            
+            // Load api state change workflow configurations
+            workflowConfig = config.getApiStateChange();
+            executorClass = workflowConfig.getExecutor();
+            clazz = WorkflowConfigHolder.class.getClassLoader().loadClass(executorClass);
+            workFlowExecutor = (WorkflowExecutor) clazz.newInstance();
+            properties = workflowConfig.getProperty();
+            if (properties != null) {
+                loadProperties(properties, workFlowExecutor);
+            }
+            workflowExecutorMap.put(WorkflowConstants.WF_TYPE_AM_API_STATE, workFlowExecutor);
+            
+            // Load singnup workflow configurations
+            workflowConfig = config.getUserSignUp();
+            executorClass = workflowConfig.getExecutor();
+            clazz = WorkflowConfigHolder.class.getClassLoader().loadClass(executorClass);
+            workFlowExecutor = (WorkflowExecutor) clazz.newInstance();
+            properties = workflowConfig.getProperty();
+            if (properties != null) {
+                loadProperties(properties, workFlowExecutor);
+            }
+            workflowExecutorMap.put(WorkflowConstants.WF_TYPE_AM_USER_SIGNUP, workFlowExecutor);
+*/
+
+        } catch (ClassNotFoundException e) {
+            log.error("Unable to find class", e);
```

Comment:
Since handleException method logs the error , when you log the error here, it will get duplicate. 
IMHO if there is no special reason we should only log or throw the exception without doing both at the same point.